File: Chess-android-test/src/main/kotlin/Move.kt
class Move(_player: Player, _chessNotation: String) {
    val from: String
    val to: String
    val promotion: String?
    val player: Player

    init {
        from = _chessNotation.substring(0, 2)
        to = _chessNotation.substring(2, 4)
        promotion = if (_chessNotation.length == 5) _chessNotation.substring(4, 5) else null
        player = _player
    }
}

File: Chess-ios/Chess/asd/Move.swift
//
//  Move.swift
//  chess-program
//
//  Created by Cseri Ádám on 2022. 05. 02..
//

import Foundation

class Move {
    let from: String
    let to: String
    let promotion: String?
    let player: Player
    
    init(player: Player, chessNotation: String) {
        self.from = chessNotation[0..<2]
        self.to = chessNotation[2..<4]
        promotion = (chessNotation.count == 5) ? chessNotation[4..<5] : nil
        self.player = player
    }
}
